File: artwork-service/src/main/java/com/gallery/artworkservice/domain/Artwork.java
package com.gallery.artworkservice.domain;

import java.util.List;

public class Artwork {
    private ArtworkID id;
    private String title;
    private int yearCreated;
    private String type;
    private int artistId;
    private List<ArtworkImage> images;




    public Artwork() {
        this.id = new ArtworkID();
        this.title = "";
        this.yearCreated = 0;
        this.type = "";
        this.artistId = 0;
    }

    public Artwork(ArtworkID id, String title, int yearCreated, String type, int artistId, List<ArtworkImage> images) {
        this.id = id;
        this.title = title;
        this.yearCreated = yearCreated;
        this.type = type;
        this.artistId = artistId;
        this.images = images;
    }


    public ArtworkID getId() {
        return id;
    }

    public void setId(ArtworkID id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public int getYearCreated() {
        return yearCreated;
    }

    public void setYearCreated(int yearCreated) {
        this.yearCreated = yearCreated;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public int getArtistId() {
        return artistId;
    }

    public void setArtistId(int artistId) {
        this.artistId = artistId;
    }

    public List<ArtworkImage> getImages() {
        return images;
    }

    public void setImages(List<ArtworkImage> images) {
        this.images = images;
    }
}

File: app/ArtworkService/Domain/Artwork.cs
namespace ArtworkService.Domain
{
    public class Artwork
    {
        private ArtworkID _id;
        private string _title;
        private int _yearCreated;
        private string _type;
        private int _artistId;

        private double _price;

        public Artwork()
        {
            this._id = new ArtworkID();
            this._title = "";
            this._yearCreated = 0;
            this._type = "";
            this._artistId = 0;
            this._price = 0.0;
        }

        public Artwork(ArtworkID id, string? title, int yearCreated, string? type, int artistId,double price)
        {
            this._id = id;
            this._title = title != null ? title : "";
            this._yearCreated = yearCreated;
            this._type = type != null ? type : "";
            this._artistId = artistId;
            this._price = price;
        }

        public Artwork(int artworkId, string? title, int yearCreated, string? type, int artistId,double price)
        {
            this._id = new ArtworkID(artworkId);
            this._title = title != null ? title : "";
            this._yearCreated = yearCreated;
            this._type = type != null ? type : "";
            this._artistId = artistId;
            this._price = price;
        }

        public Artwork(Artwork artwork)
        {
            this._id = new ArtworkID(artwork._id);
            this._title = artwork._title;
            this._yearCreated = artwork._yearCreated;
            this._type = artwork._type;
            this._artistId = artwork._artistId;
            this._price = artwork._price;
        }

        public ArtworkID Id
        {
            get { return this._id; }
            set { this._id = value; }
        }

        public string Title
        {
            get { return this._title; }
            set { this._title = value; }
        }

        public int YearCreated
        {
            get { return this._yearCreated; }
            set { this._yearCreated = value; }
        }

        public string Type
        {
            get { return this._type; }
            set { this._type = value; }
        }

        public int ArtistId
        {
            get { return this._artistId; }
            set { this._artistId = value; }
        }

        public double Price
        {
            get { return this._price; }
            set { this._price = value; }
        }
    }
}
